src/java/htsjdk/samtools/CRAMIterator.java

```diff
@@ -165,6 +165,7 @@ private void nextContainer() throws IOException, IllegalArgumentException,
 
         if (container.sequenceId == SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX) {
             refs = new byte[]{};
+            prevSeqId = -1;
         } else if (container.sequenceId == -2) {
             refs = null;
             prevSeqId = -2;
```

Comment:
We should replace magic numbers with constants. In this case its hard to tell what the value -2 means, or what namespace the sequenceID is drawn from.